### PR TITLE
Task 4 -done

### PR DESCRIPTION
**Pull request by terapro**

### CODE REVIEW

File: src/components/body/body.test.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    const wrapper = shallow(<Body searchPhrase ={mockProps.searchPhrase}
+                                  filmModeGenre={mockProps.filmModeGenre}
+                                  startSearch={mockProps.startSearch}
+                                  searchResult={mockProps.searchResult}
+                                  setFilmMode={mockProps.setFilmMode}
+                                  searchMode={mockProps.searchMode}
+                                  filmMode={mockProps.filmMode}
+                                  filmInfo={mockProps.filmInfo}
+                                  setSearchModeCallback={mockProps.setSearchModeCallback}
+                                  searchInFilmModeByGenre={mockProps.searchInFilmModeByGenre} />);
```
Comment (alex-belmach) [inline src/components/body/body.test.js]: looks like it's `<Body {...mockProps} />`

File: src/components/body/body.test.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    expect(wrapper).toMatchSnapshot();
+  });
+});
```
Comment (alex-belmach) [inline src/components/body/body.test.js]: So many props and only 1 test? Seems like we can add more

File: src/components/common/input/input.test.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+describe('<Input />', () => {
+  describe('Rendering', () => {
+    it('should render the component', () => {
+      const wrapper = shallow(<Input
+        searchInputCallback = {mockFn}
+        onKeyPressedFunc = {mockText}
+      />);
+      expect(wrapper).toMatchSnapshot();
+    });
+  });
```
Comment (alex-belmach) [inline src/components/common/input/input.test.js]: More tests could be added. E.g. testing placeholder

File: src/components/common/title/title.test.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+describe('<Title />', () => {
+  describe('Rendering', () => {
+    it('should render the component', () => {
+      const wrapper = shallow(<Title content={mockContent} />);
+      expect(wrapper).toMatchSnapshot();
+    });
+    it('should render the component with default content', () => {
+      const wrapper = shallow(<Title content={mockContent} />);
+      expect(wrapper).toMatchSnapshot();
+    });
```
Comment (alex-belmach) [inline src/components/common/title/title.test.js]: Tests are the same

File: src/components/film/film-header/film-header.test.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+import {FilmHeader} from 'src/components/film/film-header/film-header.js';
+import {shallow} from 'enzyme';
+
+const {title, rating} = {
```
Comment (alex-belmach) [inline src/components/film/film-header/film-header.test.js]: `const title = ...;
const rating = ...;` will be better

File: src/components/footer/footer.test.js [hunk 6]
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
+import {Footer} from 'src/components/footer/footer.js';
+import {shallow} from 'enzyme';
+
+describe('<Footer />', () => {
+  describe('<Footer />', () => {
```
Comment (alex-belmach) [inline src/components/footer/footer.test.js:6]: no need to repeat

File: src/components/summary/film-sort/__snapshots__/film-sort.test.js.snap [hunk 7]
```diff
@@ -0,0 +1,3 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`<FilmSort /> Render should render the component in the film mode 1`] = `""`;
```
Comment (alex-belmach) [inline src/components/summary/film-sort/__snapshots__/film-sort.test.js.snap]: Is it expected result?

File: src/components/body/body.test.js (hunk 1 above)
Comment (terapro) [reply src/components/body/body.test.js]: ok.

File: src/components/body/body.test.js (hunk 2 above)
Comment (terapro) [reply src/components/body/body.test.js]: more tests are added

File: src/components/common/input/input.test.js (hunk 3 above)
Comment (terapro) [reply src/components/common/input/input.test.js]: more tests are added

File: src/components/common/title/title.test.js (hunk 4 above)
Comment (terapro) [reply src/components/common/title/title.test.js]: corrected

File: src/components/film/film-header/film-header.test.js (hunk 5 above)
Comment (terapro) [reply src/components/film/film-header/film-header.test.js]: ok

File: src/components/footer/footer.test.js (hunk 6 above)
Comment (terapro) [reply src/components/footer/footer.test.js:6]: corrected

File: src/components/summary/film-sort/__snapshots__/film-sort.test.js.snap (hunk 7 above)
Comment (terapro) [reply src/components/summary/film-sort/__snapshots__/film-sort.test.js.snap]: no
.. fixed

**Pull request merged**